### ISSUE TITLE
Conditionally show points

### ISSUE THREAD

**Issue by iocube** [c3js/c3#2054]
Hello.

`point.show` is all or nothing. I can't control which points I want to show / hide.
What if it will be possible to pass a function which should return true / false, and based on that, we'll show or hide the point.

Basically, I am talking about modifying `core.js#c3_chart_internal_fn.opacityForCircle` to something like this:

```javascript
c3_chart_internal_fn.opacityForCircle = function (d) {
    var isPointShouldBeShown;
    if (this.config.point_show instanceof Function) {
        isPointShouldBeShown = this.config.point_show(d);
    } else {
        isPointShouldBeShown = this.config.point_show;
    }
    var opacity = isPointShouldBeShown ? 1 : 0;
    return isValue(d.value) ? (this.isScatterType(d) ? 0.5 : opacity) : 0;
};
```

So the usage example will be:
```javascript
points: {
    show: function(d) {
        if (d.index % 2 === 0) { return true; }
        return false;
    }
}
```

or
```
points: {
    show: true
}
```
note: The old behavior is preserved.

What you guys think?

**Comment by kt3k** [comment]
Sounds like a reasonable enhancement to me. PRs are welcome.

@aendrew @masayuki0812 
How do you think?

**Comment by iocube** [comment]
A co-worker suggested that I can achieve same behavior if I will use `point.r`

Turns out `point.r` actually can be a boolean **or** a function which expects a `datapoint` as an argument (by the way, it is not documented [here](http://c3js.org/reference.html))

So if this function returns 0, then the point will not be shown.

Is it still reasonable?

**Comment by kt3k** [comment]
Thanks for pointing the above.

Personally I still feel it's reasonable to have function option for `point.show` because it feels like a hack to set radius=0 for hiding it.

However this is design issue, so I'm going to collect other maintainers' opinions somewhere later.

**Comment by kt3k** [comment]
Implemented at https://github.com/c3js/c3/pull/2062

This will be released at v0.4.13.
